core/ir.mli

```diff
@@ -59,7 +59,16 @@ and tail_computation =
   | Special    of special
   | Case       of value * (binder * computation) name_map * (binder * computation) option
   | If         of value * computation * computation
-and fun_def = binder * (tyvar list * binder list * computation) * binder option * location * bool
+and fun_def =
+  {
+    binder   : binder;
```

Comment:
Ideally each of these field names would be prefixed, e.g. `fn_binder : binder`.

Reply:
Unfortunately, these prefixes make record field punning less convenient: For example, we always use `tyvars` as the corresponding variable name, which would require `{fn_tyvar = tyvar; ... }` every single time.

Reply:
Do you (or others) feel strongly enough about this so I should change this or not?

Reply:
I think the reason for such prefixing is that in OCaml there's no way to disambiguate between field names defined in different modules that happen to both be in the same scope: whoever was declared last wins.  Hence, prefixing is a way of defensive programming around this limitation.  (I had a vague recollection that something may have been done about this in modern OCamls but can't remember how/what.)  In any case, I think it's probably better to stick with this convention uniformly even though it makes code a little more painful.

Reply:
I've update the PR to prefix all the fields with `fn_`